src/Configuration/Config.php

```diff
@@ -91,17 +91,38 @@ public function getTargetSnowflakeUser(): string
 
     public function getSourceSnowflakePassword(): string
     {
-        return $this->getStringValue(['parameters', 'credentials', 'source', '#password']);
+        return $this->getStringValue(['parameters', 'credentials', 'source', '#password'], '');
     }
 
     public function getMigrationSnowflakePassword(): string
     {
-        return $this->getStringValue(['parameters', 'credentials', 'migration', '#password']);
+        return $this->getStringValue(['parameters', 'credentials', 'migration', '#password'], '');
     }
 
     public function getTargetSnowflakePassword(): string
     {
-        return $this->getStringValue(['parameters', 'credentials', 'target', '#password']);
+        return $this->getStringValue(['parameters', 'credentials', 'target', '#password'], '');
+    }
+
+    public function getSourceSnowflakePrivateKey(): ?string
+    {
+        /** @var string|false $value */
```

Comment:
Proč ne `?string`? 🤔

Reply:
protože getValue metoda když by si chtěl vrátit null, tak to spadne na exceptioně, proto ten false a ten další řádek no